mmdet3d/core/visualizer/open3d_vis.py

```diff
@@ -0,0 +1,512 @@
+            the color of points which are in bbox3d.
+        rot_axis (int): rotation axis of bbox.
+        center_mode (str): indicate the center of bbox is
+            bottom center or gravity center.
+    """
+
+    def __init__(self,
+                 points,
+                 bbox3d=None,
+                 mode='xyz',
+                 save_path=None,
+                 points_size=2,
+                 point_color=(0.5, 0.5, 0.5),
+                 bbox_color=(0, 1, 0),
+                 points_in_box_color=(1, 0, 0),
+                 rot_axis=2,
+                 center_mode='lidar_bottom'):
+        super(Visualizer, self).__init__()
+        assert 0 <= rot_axis <= 2
+
+        # init visualizer
+        self.o3d_visualizer = o3d.visualization.Visualizer()
+        self.o3d_visualizer.create_window()
+        mesh_frame = geometry.TriangleMesh.create_coordinate_frame(
+            size=1, origin=[0, 0, 0])  # create coordinate frame
+        self.o3d_visualizer.add_geometry(mesh_frame)
+
+        self.points_size = points_size
+        self.point_color = point_color
+        self.bbox_color = bbox_color
+        self.points_in_box_color = points_in_box_color
+        self.rot_axis = rot_axis
+        self.center_mode = center_mode
+        self.mode = mode
+
+        # draw points
+        if points is not None:
+            self.pcd, self.points_colors = _draw_points(
+                points, self.o3d_visualizer, points_size, point_color, mode)
+
+        # draw boxes
+        if bbox3d is not None:
+            _draw_bboxes(bbox3d, self.o3d_visualizer, self.points_colors,
+                         self.pcd, bbox_color, points_in_box_color, rot_axis,
+                         center_mode, mode)
+
+    def add_bboxes(self, bbox3d, bbox_color=None, points_in_box_color=None):
+        """Add bounding box to visualizer.
+
+        Args:
+            bbox3d (numpy.array, shape=[M, 7]):
+                3d bbox(x, y, z, dx, dy, dz, yaw) to visualize. The 3d bbox
+                is in mode of Box3DMode.DEPTH with
+            gravity_center (please refer to core.structures.box_3d_mode)
+            bbox_color (tuple[float]): the color of bbox.
+            points_in_box_color (tuple[float]): the color of points which
+                are in bbox3d.
+        """
+        if bbox_color is None:
+            bbox_color = self.bbox_color
+        if points_in_box_color is None:
+            points_in_box_color = self.points_in_box_color
+        _draw_bboxes(bbox3d, self.o3d_visualizer, self.points_colors, self.pcd,
+                     bbox_color, points_in_box_color, self.rot_axis,
+                     self.center_mode, self.mode)
+
+    def show(self, save_path=None):
+        """Visualize the points cloud."""
+        self.o3d_visualizer.run()
+
+        if save_path is not None:
+            self.o3d_visualizer.capture_screen_image(save_path)
+
+        self.o3d_visualizer.destroy_window()
+        return
+
+
+if __name__ == '__main__':
```

Comment:
Should remove this part during merge as we do not wish it can be used as a script. But we may add a script in tools for browsing datasets.